Formula/postgresql@14.rb

```diff
@@ -104,41 +108,90 @@ def post_install
   end
 
   def postgresql_datadir
+    if old_postgres_data_dir.exist?
+      old_postgres_data_dir
+    else
+      var/name
+    end
+  end
+
+  def pg_version_exists?
+    (postgresql_datadir/"PG_VERSION").exist?
+  end
+
+  def old_postgres_data_dir
     var/"postgres"
   end
 
-  def postgresql_log_path
-    var/"log/postgres.log"
+  def postgresql_formula_present?
+    Formula["postgresql"].any_version_installed?
   end
 
-  def pg_version_exists?
-    (postgresql_datadir/"PG_VERSION").exist?
+  # Figure out what version of PostgreSQL the old data dir is
+  # using
+  def old_postgresql_datadir_version
+    pg_version = old_postgres_data_dir/"PG_VERSION"
+    pg_version.exist? && pg_version.read.chomp
   end
 
   def caveats
-    <<~EOS
-      To migrate existing data from a previous major version of PostgreSQL run:
-        brew postgresql-upgrade-database
+    caveats = ""
+
+    # Extract the version from the formula name
+    pg_formula_version = name.split("@", 2).last
+    # ... and check it against the old data dir postgres version number
+    # to see if we need to print a warning re: data dir
+    if old_postgresql_datadir_version == pg_formula_version
+      caveats += if postgresql_formula_present?
+        # Both original PostgreSQL and PostgreSQL@14 are installed
+        <<~EOS
+          Previous versions of this formula used the same data directory as
+          the regular PostgreSQL formula. This causes a conflict if you
+          try to use both at the same time.
+
+          In order to avoid this conflict, you should make sure that the
+          #{name} data directory is located at:
+            #{postgresql_datadir}
+
+        EOS
+      else
+        # Only PostgreSQL@14 is installed, not original PostgreSQL
+        <<~EOS
+          Previous versions of #{name} used the same data directory as
+          the postgresql formula. This will cause a conflict if you
+          try to use both at the same time.
+
+          You can migrate to a versioned data directory by running:
+            mv -v "#{old_postgres_data_dir}" "#{postgresql_datadir}"
+
+          (Make sure PostgreSQL is stopped before executing this command)
+
+        EOS
+      end
+    end
```

Comment:
Copied from `postgresql@13`. This is perhaps not the logic we want now. I'd love thoughts here.

Reply:
Makes sense to me, but I don't use postgres.